tests/consistency/COVERAGE.md

```diff
@@ -0,0 +1,817 @@
+# Test coverage of SpeechBrain
```

Comment:
Adel linked me to this.  
I did read it all earlier today but I got distracted with the ksponspeech debugging :)

I think it's good overall and nothing really seemed weird to me but I might not be the best person to ask since I don't have deep knowledge of all the SB internals, or just on CI/CD in general.

I don't know who exactly the document is destined for, but if it's intended both for core contributors and occasional contributors, I feel like it might not feel enough like a practical guide for the latter.  
Also the references to other parts of the documents makes it a bit painful to navigate overall, but I'm not sure what's the best way to remedy that.

Reply:
The references within are an issue. What just crossed my mind is making a directory which has parts of this split up - and then cross-linking them.

The use groups is a good point - right now it is more written as an "informal proof" to have a tool (a first draft) for show-casing that this is the case and to which extent. How to write proofs that get read is next ;-)